runhouse/resources/functions/aws_lambda.py

```diff
@@ -149,10 +162,219 @@ def from_name(cls, name, dryrun=False, alt_options=None):
+        cls, paths_to_code, env, runtime, timeout, memory_size, tmp_size, retention_time
+    ):
+
+        if isinstance(env, str) and "requirements.txt" in env:
+            paths_to_code.append(Path(env).absolute())
+
+        if runtime is None or runtime not in LambdaFunction.SUPPORTED_RUNTIMES:
+            warnings.warn(
+                f"{runtime} is not a supported by AWS Lambda. Setting runtime to python3.9."
+            )
+            runtime = LambdaFunction.DEFAULT_PY_VERSION
+
+        if timeout is None:
+            warnings.warn("Timeout set to 15 min.")
+            timeout = LambdaFunction.DEFAULT_TIMEOUT
+        else:
+            if (env.reqs is not None or len(env.reqs) > 0) and timeout < 600:
+                warnings.warn(
+                    "Increasing the timeout to 600 sec, in order to enable the packages setup."
+                )
+                timeout = 600
+            if timeout > 900:
+                timeout = 900
+                warnings.warn(
+                    "Timeout can not be more then 900 sec, setting to 900 sec."
+                )
+            if timeout < 3:
+                timeout = 3
+                warnings.warn("Timeout can not be less then 3 sec, setting to 3 sec.")
+        if memory_size is None:
+            warnings.warn("Memory size set to 1024 MB.")
+            memory_size = LambdaFunction.DEFAULT_MEMORY_SIZE
+        else:
+            if (
+                env.reqs is not None or len(env.reqs) > 0
+            ) and memory_size < LambdaFunction.DEFAULT_MEMORY_SIZE:
+                warnings.warn(
+                    "Increasing the memory size to 1GB, in order to enable the packages setup."
+                )
+                memory_size = LambdaFunction.DEFAULT_MEMORY_SIZE
+            if memory_size < 128:
+                memory_size = 128
+                warnings.warn(
+                    "Memory size can not be less then 128 MB, setting to 128 MB."
+                )
+            if memory_size > 10240:
+                memory_size = 10240
+                warnings.warn(
+                    "Memory size can not be more then 10240 MB, setting to 10240 MB."
+                )
+        if tmp_size is None or tmp_size < LambdaFunction.DEFAULT_TMP_SIZE:
+            warnings.warn(
+                "Setting /tmp size to 3GB, in order to enable the packages setup."
+            )
+            tmp_size = LambdaFunction.DEFAULT_TMP_SIZE
+        elif tmp_size > 10240:
+            tmp_size = 10240
+            warnings.warn(
+                "/tmp size can not be more then 10240 MB, setting to 10240 MB."
+            )
+        if retention_time is None:
+            retention_time = LambdaFunction.DEFAULT_RETENTION
+
+        return (
+            paths_to_code,
+            env,
+            runtime,
+            timeout,
+            memory_size,
+            tmp_size,
+            retention_time,
+        )
+
     def _lambda_exist(self, name):
```

Comment:
instead of returning `True` return the function itself? that way we save an extra call to that same API later on